src/libstd/ffi/c_str.rs

```diff
@@ -328,6 +350,53 @@ impl CStr {
     pub fn to_bytes_with_nul(&self) -> &[u8] {
         unsafe { mem::transmute::<&[libc::c_char], &[u8]>(&self.inner) }
     }
+
+    /// Yields a `&str` slice if the `CStr` contains valid UTF-8.
+    ///
+    /// This function will calculate the length of this string and check for
+    /// UTF-8 validity, and then return the `&str` if it's valid.
+    ///
+    /// > **Note**: This method is currently implemented to check for validity
+    /// > after a 0-cost cast, but it is planned to alter its definition in the
+    /// > future to perform the length calculation in addition to the UTF-8
+    /// > check whenever this method is called.
+    #[unstable(feature = "cstr", reason = "recently added")]
+    pub fn to_str(&self) -> Result<&str, str::Utf8Error> {
+        // NB: When CStr is changed to perform the length check in .to_bytes() instead of in
+        // from_ptr(), it may be worth considering if this should be rewritten to do the UTF-8
+        // check inline with the length calculation instead of doing it afterwards.
+        str::from_utf8(self.to_bytes())
+    }
+
+    /// Yields a `&str` slice without checking if the `CStr` contains valid UTF-8.
+    ///
+    /// This function will calculate the length of this string (which normally
+    /// requires a linear amount of work to be done) and then return the
+    /// resulting slice as a `&str` without checking for UTF-8 validity.
+    ///
+    /// > **Note:** This method is currently implemented as a 0-cost cast, but
+    /// > it is planned to alter its definition in the future to perform the
+    /// > length calculation whenever this method is called.
+    #[unstable(feature = "cstr", reason = "recently added")]
+    pub unsafe fn to_str_unchecked(&self) -> &str {
```

Comment:
Conventionally these sorts of convenience methods are not found in the standard library, as composition of primitives is generally favored above having a matrix of methods. We do have some counterexamples to this, but in each case it's generally because the method is used 99% of the time basically. I'm not sure that `to_str_unchecked` falls within that category personally.